src/editor/codemirror/dnd.ts

```diff
@@ -30,7 +30,17 @@ interface LastDragPos {
   previewUndo: ChangeSet;
 }
 
-let draggedCode: string | undefined;
+export interface DragContext {
+  code: string;
+  // I'd like these to be "example" and "call"
+  // "example" just inserts code as it always has and is used when it's Explore or a non-function call from Reference
+  // "call" is used for function calls and results in cursor between the brackets and signature help
+  // single line examples have the cursor at the end of the line
+  // multi-line examples have the cursor at the beginning of the example (ideally ignoring whitespace but can do that later)
```

Comment:
Next steps as discussed.

Also look at testing the calculateChanges behaviour including the selection behaviour.